packages/eslint-plugin-qwik/src/noSyncClosure.ts

```diff
@@ -0,0 +1,48 @@
+const { Rule } = require('eslint');
+
+const isBrowserAPI = (variable, context) => {
+  const browserAPIs = new Set([
+    'window', 'document', 'navigator', 'location', 'history', 'localStorage',
+    'sessionStorage', 'alert', 'fetch', // ... other known browser APIs
+  ]);
+
+  return context.getScope().through.some(scopeVar => 
+    scopeVar.identifier.name === variable && browserAPIs.has(variable)
```

Comment:
Won't this run into trouble when you do `{ const document = "foo"; sync$(() => document)}`?

Since all these are global and globals aren added to the scope array ([see the bundles here](https://qwik.dev/playground/#f=Q0405gbYEo2OAijzUjHtgCrMxLwAYOGhYlsNNhuoCqgoBZor9BxDQoI8nUJDXOP9%2FF1cNWtHE9swSWwA)), would it instead be possible to just check if it's using only globals?

You could even check if they're browser globals but since it would crash immediately for undefined globals I think that's not necessary.